src/coreclr/tools/Common/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -371,7 +371,19 @@ private MethodSignature ParseMethodSignatureImpl(bool skipEmbeddedSignatureData)
                 Debug.Assert((int)MethodSignatureFlags.CallingConventionVarargs == (int)SignatureCallingConvention.VarArgs);
                 Debug.Assert((int)MethodSignatureFlags.UnmanagedCallingConvention == (int)SignatureCallingConvention.Unmanaged);
 
-                flags = (MethodSignatureFlags)signatureCallConv;
+                // If skipEmbeddedSignatureData is true, we're building the signature for the purposes of building a type.
+                // We normalize unmanaged calling convention into a single value - "unmanaged".
```

Comment:
Not sure I understand why this is safe. You mean that, after the other change, it no longer matters if you have

`delegate* unmanaged<int> p`

vs. 

`delegate* unmanaged<[SomeModReq]int> p`

We effectively erase custom modifiers for unmanaged function pointers?

Reply:
Modifiers were already erased for runtime pointer __types__ - that's what the existing `skipEmbeddedSignatureData` argument controls. It distinguishes between "are we parsing a standalone method signature" vs "are we parsing a TypeSpec that is a function pointer typespec". Custom modifiers never matter for type identity - modifiers only matter for signature resolution, but they don't contribute to identity.

We have some tests testing this like:

https://github.com/dotnet/runtime/blob/ed0f1c5d128d9d58079cc706f3e75dedbe80459b/src/tests/reflection/Modifiers/modifiersdata.il#L160-L166

The logic I'm adding basically switches specific calling conventions into things that act like modifiers (previously they would be part of the type identity).

The added unit test is then testing two things:
1. That we still consider signatures that differ by modopts to be different signatures
2. The fact that asking for the type of the method parameter only distinguishes managed/unmanaged